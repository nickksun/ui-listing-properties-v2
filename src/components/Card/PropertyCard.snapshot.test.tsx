import React from "react";
import PropertyCard from "./";
import renderer from "react-test-renderer";
import CARD_STATUS from "../../constants/cardStatusConstants";

const mockProperty = {
  price: "$526,500",
  agency: {
    brandingColors: {
      primary: "#000000",
    },
    logo:
      "https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif",
  },
  id: "4",
  mainImage:
    "https://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
};

describe("Property Card render", () => {
  it("renders ADD Card correctly", () => {
    const cb = jest.fn();
    const tree = renderer
      .create(
        <PropertyCard
          isSaved={false}
          buttonText={CARD_STATUS.ADD_PROPERTY}
          key={mockProperty.id}
          id={mockProperty.id}
          price={mockProperty.price}
          mainImage={mockProperty.mainImage}
          agency={mockProperty.agency}
          onChange={cb}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it("renders SAVED Card correctly", () => {
    const cb = jest.fn();
    const tree = renderer
      .create(
        <PropertyCard
          isSaved={true}
          buttonText={CARD_STATUS.SAVED_PROPERTY}
          key={mockProperty.id}
          id={mockProperty.id}
          price={mockProperty.price}
          mainImage={mockProperty.mainImage}
          agency={mockProperty.agency}
          onChange={cb}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders REMOVE Card correctly", () => {
    const cb = jest.fn();
    const tree = renderer
      .create(
        <PropertyCard
          buttonText={CARD_STATUS.REMOVE_PROPERTY}
          key={mockProperty.id}
          id={mockProperty.id}
          price={mockProperty.price}
          mainImage={mockProperty.mainImage}
          agency={mockProperty.agency}
          onChange={cb}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
