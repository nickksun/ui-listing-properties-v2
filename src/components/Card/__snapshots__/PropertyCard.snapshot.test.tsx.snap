// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Card render renders ADD Card correctly 1`] = `
<div
  className="sc-gsTCUz fcOIGg card"
>
  <div
    className="sc-dlfnbm jrEzrk card-header"
    color="#000000"
  >
    <img
      alt="agency logo"
      src="https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif"
    />
  </div>
  <img
    alt="property image"
    className="card-img-top"
    src="https://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg"
  />
  <div
    className="sc-hKgILt ezlxUD card-body"
  >
    <div>
      $526,500
    </div>
    <button
      className="sc-bdfBwQ dVeWDV btn btn-success"
      disabled={false}
      onClick={[MockFunction]}
      test-attr="button"
      type="button"
    >
      Add
    </button>
  </div>
</div>
`;

exports[`Property Card render renders REMOVE Card correctly 1`] = `
<div
  className="sc-gsTCUz fcOIGg card"
>
  <div
    className="sc-dlfnbm jrEzrk card-header"
    color="#000000"
  >
    <img
      alt="agency logo"
      src="https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif"
    />
  </div>
  <img
    alt="property image"
    className="card-img-top"
    src="https://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg"
  />
  <div
    className="sc-hKgILt ezlxUD card-body"
  >
    <div>
      $526,500
    </div>
    <button
      className="sc-bdfBwQ dVeWDV btn btn-success"
      disabled={false}
      onClick={[MockFunction]}
      test-attr="button"
      type="button"
    >
      Remove
    </button>
  </div>
</div>
`;

exports[`Property Card render renders SAVED Card correctly 1`] = `
<div
  className="sc-gsTCUz fcOIGg card"
>
  <div
    className="sc-dlfnbm jrEzrk card-header"
    color="#000000"
  >
    <img
      alt="agency logo"
      src="https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif"
    />
  </div>
  <img
    alt="property image"
    className="card-img-top"
    src="https://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg"
  />
  <div
    className="sc-hKgILt ezlxUD card-body"
  >
    <div>
      $526,500
    </div>
    <button
      className="sc-bdfBwQ dVeWDV btn btn-secondary"
      disabled={true}
      onClick={[MockFunction]}
      test-attr="button"
      type="button"
    >
      Saved
    </button>
  </div>
</div>
`;
